compiler/src/dotty/tools/dotc/transform/TypeTestsCasts.scala

```diff
@@ -96,23 +96,30 @@ trait TypeTestsCasts {
         /** Transform isInstanceOf OrType
          *
          *    expr.isInstanceOf[A | B]  ~~>  expr.isInstanceOf[A] | expr.isInstanceOf[B]
+         *    expr.isInstanceOf[A & B]  ~~>  expr.isInstanceOf[A] & expr.isInstanceOf[B]
```

Comment:
An identical commit has now been merged as part of https://github.com/lampepfl/dotty/pull/2082